python/cuml/common/base.pyx

```diff
@@ -152,3 +207,31 @@ class Base:
     def __setstate__(self, state):
         self.__dict__.update(state)
         self.handle = cuml.common.handle.Handle()
+
+    def __getattr__(self, attr):
+        real_name = '_' + attr
+        if real_name in self.__dict__.keys():
+            if isinstance(self.__dict__[real_name], Array):
+                return self.__dict__[real_name].to_output(self.output_type)
+
+    def _get_output_type(self, output_dtype):
+        if isinstance(output_dtype, str):
+            output_type = output_dtype.lower()
+            if output_type in ['numpy', 'cupy', 'cudf', 'numba']:
+                return output_dtype
+        else:
+            self.output_type = self.output_type
+
+    def _check_output_type(self, input):
+        """
+        Method to be called by fit/predict/etc. method of inheriting classes
```

Comment:
see question earlier... also, to me this should be called `_configure_output_type` since it mutates self. `check_` sounds like it just asserts on errors.